src/ai/core/signal_processor.py

```
@@ -0,0 +1,172 @@
+from typing import Dict, List, Optional
+from dataclasses import dataclass
+import numpy as np
```

Ruff (0.8.2)

`numpy` imported but unused

Remove unused import: `numpy`

(F401)